src/BeliefTypes.jl

```diff
@@ -41,3 +41,49 @@ mutable struct NBPMessage <: Singleton
 end
 
 ### SOME CONVERGENCE REQUIRED ^^^
+
+#DEV NOTE it looks like it can be consolidated into one type
+# if we can pass messages similar to EasyMessage:
+# pts::Array{Float64,2}
+# bws::Array{Float64,1}
+# option a tuple
+# bellief::Dict{Symbol, NamedTuple{(:vec, :bw, :inferdim),Tuple{Array{Int64,1},Array{Int64,1},Float64}}}
+# or an extra type
+# or the MsgPrior/PackedMessagePrior, depending on the serialization requirement of the channel
+# but I would think only one message type
+
+# mutable struct NBPMessage <: Singleton
+#   status::Symbol # Ek kort die in die boodskap
+#   p::Dict{Symbol, EasyMessage}
+# end
+
+struct TreeBelief
+  val::Array{Float64,2}
+  bw::Array{Float64,2}
+  inferdim::Float64
+  # manifolds::T TODO ? JT not needed as you have the variable with all the info in it?
```

Comment:
if `manifolds` was in the original struct then its probably definitely needed, please add.  We can do consolidation and reduction separately.  This field is probably used in a situation when the variable data is not readily available.

Reply:
Manifold is not in the current mm implementation. 
`TempBeliefMsg = Dict{Symbol, Tuple{BallTreeDensity, Float64}}`

Reply:
can't remember immediately but I do recall something about any code using `TempBeliefMsg` having to go back to a/the factor graph to `getManifolds` using the provided `Symbol` -- it was a schlep because it required a valid `dfg` object to be passed all the way down to that function...

Reply:
Lets continue this discussion in #459

Reply:
Its in struct, but not populated everywhere yet.